core/arch/riscv/include/kernel/clint.h

```diff
@@ -35,12 +35,12 @@ static inline void clint_ipi_clear(unsigned long hart)
 static inline void clint_set_mtimecmp(uint64_t timecmp)
 {
 	/* Each hart has a separate source of timer interrupts */
-	io_write64(CLINT_MTIMECMP(get_core_pos()), timecmp);
+	io_write64(CLINT_MTIMECMP(get_core_pos_hartid()), timecmp);
```

Comment:
I don't think we want to use hart ID here.
The RISC-V Advanced Core Local Interruptor Specification uses "HART index".